Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClientContextProject.cs

```diff
@@ -28,29 +29,35 @@ internal sealed class PythonLanguageClientContextProject : IPythonLanguageClient
         public event EventHandler InterpreterChanged;
         public event EventHandler SearchPathsChanged;
         public event EventHandler Closed;
+        public event EventHandler ReanalyzeChanged;
 
         public PythonLanguageClientContextProject(PythonProjectNode project) {
             _project = project ?? throw new ArgumentNullException(nameof(project));
             _disposables = new DisposableBag(GetType().Name);
 
             _project.LanguageServerInterpreterChanged += OnInterpreterChanged;
             _project.LanguageServerSearchPathsChanged += OnSearchPathsChanged;
+            _project.ReanalyzeProject_Notify += OnReanalyzeProject;
             _disposables.Add(() => {
                 _project.LanguageServerInterpreterChanged -= OnInterpreterChanged;
                 _project.LanguageServerSearchPathsChanged -= OnSearchPathsChanged;
+                _project.ReanalyzeProject_Notify -= OnReanalyzeProject;
             });
 
             _project.AddActionOnClose(this, (obj) => Closed?.Invoke(this, EventArgs.Empty));
         }
 
         public InterpreterConfiguration InterpreterConfiguration => _project.ActiveInterpreter?.Configuration;
 
-        public string RootPath => _project.ProjectHome;
+        // Normalize to maintain trailing folder slashes
+        public string RootPath => CommonUtils.NormalizeDirectoryPath(_project.ProjectHome);
```

Comment:
there was an issue where the LSP client code sending a different rootPath so now we normalize so that we match